discovery_engine_core/providers/src/client.rs

```diff
@@ -219,10 +199,13 @@ impl Client {
     }
 
     /// Run a query for fetching `Article`s from Newscatcher.
-    pub async fn query_articles(&self, query: &impl Query) -> Result<Vec<Article>, Error> {
-        self.query_newscatcher(query)
+    pub async fn query_articles(&self, query: &impl Query) -> Result<Vec<GenericArticle>, Error> {
+        let articles = self
+            .query_newscatcher(query)
             .await
-            .map(|news| news.articles)
+            .map(|news| news.articles)?;
+
+        articles.into_iter().map(GenericArticle::try_from).collect()
```

Comment:
🔧 We can't use collect here. If one news catcher article fails the conversion because of the `Url::parse` errors that have been moved from the `impl TryFrom<GenericArticle> for NewsResource` to the `impl TryFrom<NewscatcherArticle> for GenericArticle`, the iteration stops and all valid articles are discarded. Before it would have failed in `impl TryFrom<(GenericArticle, StackId, Embedding)> for Document`, allowing us to separate the valid and invalid items in the `documentify_articles` function.